### PR TITLE
[docs] Add authentication against private Git repositories

### PR DESCRIPTION
**Pull request by jorgegarciarey**
This commit adds documentation for using perceval against
private git repositories. The README.md and the read the docs
is updated with an example.

Signed-off-by: Jorge García Rey <jorgegar@inditex.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -204,6 +204,18 @@ this backend Git program has to be installed on your system.
 $ perceval git 'https://github.com/chaoss/grimoirelab-perceval.git' --from-date '2016-01-01'
 ```
 
+If you need to execute this command against a **private Git repository**, so, you must set the `username` and `password`:
```
Comment (vchrombie) [inline README.md:207]:
```suggestion
To run the backend against a **private git repository**, you must pass the credentials directly in the URL:
```

File: README.md [hunk 2]
```diff
@@ -204,6 +204,18 @@ this backend Git program has to be installed on your system.
...
+
+```
+perceval git https://username:password@repository
```
Comment (vchrombie) [inline README.md:210]:
```suggestion
$ perceval git https://username:key@repository-url
```

File: README.md [hunk 3]
```diff
@@ -204,6 +204,18 @@ this backend Git program has to be installed on your system.
...
+```
+
+For example, for private GitHub repositories:
+
+```
+perceval git https://username:password@github.com/organization/repository.git
```
Comment (vchrombie) [inline README.md:216]:
```suggestion
$ perceval git https://username:api_token@github.com/chaoss/grimoirelab-private
```

File: docs/perceval/git.md [hunk 4]
```diff
@@ -7,6 +7,17 @@ maintains data for tracking what changed, and some metadata about who committed
 change, when, which files were affected, etc. Perceval retrieves this information, and
 produces a JSON document (a dictionary when using it from Python) for each commit.
 
+### Fetch from private Git repositories
```
Comment (vchrombie) [inline docs/perceval/git.md:10]: Fetch commits from private Git repositories

File: docs/perceval/git.md [hunk 5]
```diff
@@ -7,6 +7,17 @@ maintains data for tracking what changed, and some metadata about who committed
...
+
+If you need to fech any data from a private Git repository, you must set the `username` and
+`password/api-token`.
```
Comment (vchrombie) [inline docs/perceval/git.md:13]:
```suggestion
If you want to fetch any data from a private git repository, you must pass the credentials 
(`username` and `password`/`api_token`) directly in the URL
```

File: docs/perceval/git.md [hunk 6]
```diff
@@ -7,6 +7,17 @@ maintains data for tracking what changed, and some metadata about who committed
...
+
+For example, for private GitHub repositories:
+
+```
+perceval git https://username:password@github.com/organization/respository.git
```
Comment (vchrombie) [inline docs/perceval/git.md:18]:
```suggestion
$ perceval git https://username:api_token@github.com/chaoss/grimoirelab-private
```

**Review by vchrombie — CHANGES_REQUESTED** [review]
Thanks for the PR @jorgegarciarey.

The changes look great, I completed it by adding some suggestions. Please let me know what do you think.

**Comment by vchrombie** [comment]
Please continue adding commits to this branch. Once it is finalized, we can sqaush them into one for merging in the last.

File: README.md (hunk 3 above)
Comment (jorgegarciarey) [reply README.md]: Done at https://github.com/chaoss/grimoirelab-perceval/pull/763/commits/526afd7d1fded8b702e5cbe752e2bd1160042b8a

**Comment by jorgegarciarey** [comment]
Hi @vchrombie , thanks for the pointers.
I've just updated the pull request, could you take look?

**Review by vchrombie — APPROVED** [review]
LGTM!
Thanks for working on the changes @jorgegarciarey.

Could you please squash the commits into one and write a small commit description so that I can merge the PR.

It can be something like this
```
[docs] Add information for using perceval against private git repos

This commit adds documentation for using perceval against
private git repositories. The README.md and the read the docs
is updated with an example.

<sign off>
```

Please let me know if you need any help.

**Comment by jorgegarciarey** [comment]
Ei @vchrombie, I've just squashed the 6 commits with rebase. It's ok?

**Comment by vchrombie** [comment]
> Ei @vchrombie, I've just squashed the 6 commits with rebase. It's ok?

I still see 8 commits in the PR.

No problem @jorgegarciarey, I can squash them myself and merge them. I will prolly do that.

I'm putting the steps here, for future reference
```
$ git reset --soft HEAD~n && git commit -s
$ git push -f origin master
```
here, `n` is the number of commits to be squashed

Thanks.

**Comment by jorgegarciarey** [comment]
Ei @vchrombie ,
I think I got confused and squashed more than 8 commits, sorry!

I've just opened a new pull request for this: https://github.com/chaoss/grimoirelab-perceval/pull/764

Thanks!

**Comment by vchrombie** [comment]
> I think I got confused and squashed more than 8 commits, sorry!
> 
> I've just opened a new pull request for this: #764

Oops, I think I squashed them and forced pushed to the master branch. I was not looking here.

**Pull request closed without merging**